src/GdbCommand.h

```diff
@@ -27,28 +28,146 @@ class GdbCommand {
   virtual std::string invoke(GdbServer& gdb_server, Task* t,
                              const std::vector<std::string>& args) = 0;
 
+  /**
+   * When call, gdb will automatically fill argument to this command
+   * by running gdb.execute(). This is useful to gdb state along side
+   * of the command invocation.
```

Comment:
Ah, this is just to automatically parse the arguments vector to a typed parameter list. I honestly don't think it's worth it, certainly not with the commands we currently have.

If we end up with repetitive parsing code later, I'd suggest creating a varargs template `parse_to_tuple<T1, T2, ...>(const vector<string>& args)" that commands can call explicitly if they want to.